src/Container.php

```diff
@@ -8,6 +8,7 @@
 namespace WCPay;
 
 use Psr\Container\ContainerInterface;
+use WCPay\Internal\DependencyManagement\ContainerException;
```

Comment:
Looking at https://github.com/Automattic/woocommerce-payments/pull/6862/commits/e153b8081c413db5ba676daf7fa34168a124bcad where the exception got introduced, the answer is that then Psalm exception checking was not introduced, and I didn't even realize that the League package provided exception classes.

FWIW, League has two exceptions: 

- `ContainerException extends RuntimeException implements ContainerExceptionInterface` that you are referencing.
- `NotFoundException extends InvalidArgumentException implements NotFoundExceptionInterface` which is essentially the same as our `ContainerException`.

This PR treats both the same, and converts them into an `InvalidArgumentException`, which might not be ideal, but at the same time, it might be easier to have the one `ContainerException` to worry about.

Reply:
Done in: https://github.com/Automattic/woocommerce-payments/pull/7316/commits/5a27500a11380d8d993459e2af2f0e043393d847